fil_fungible_token/src/method.rs

```diff
@@ -0,0 +1,89 @@
+use fvm_ipld_encoding::Error as IpldError;
+use fvm_ipld_encoding::RawBytes;
+use fvm_sdk::{send, sys::ErrorNumber};
+use fvm_shared::{address::Address, econ::TokenAmount, ActorID};
+use num_traits::Zero;
+use thiserror::Error;
+
+use crate::receiver::types::TokenReceivedParams;
+
+type Result<T> = std::result::Result<T, MethodCallError>;
+
+#[derive(Error, Debug)]
+pub enum MethodCallError {
+    #[error("fvm syscall error: `{0}`")]
+    Syscall(#[from] ErrorNumber),
+    #[error("ipld serialization error: `{0}`")]
+    Ipld(#[from] IpldError),
+}
+
+/// An abstraction used to send messages to other actors
+pub trait MethodCaller {
+    /// Call the receiver hook on a given actor
+    ///
+    /// Returns true if the receiver hook is called and exits without error, else returns false
+    fn call_receiver_hook(
+        &self,
+        from: ActorID,
+        to: ActorID,
+        value: &TokenAmount,
+        data: &[u8],
+    ) -> Result<bool>;
+}
+
+#[derive(Debug, Default, Clone, Copy)]
+pub struct FvmMethodCaller {}
+
+// TODO: this should be moved to the example actor so the dependency on fvm_sdk is dropped from the
+// library
+impl MethodCaller for FvmMethodCaller {
+    fn call_receiver_hook(
+        &self,
+        from: ActorID,
+        to: ActorID,
+        value: &TokenAmount,
```

Comment:
This initially confused me (I thought it was going to transfer native FIL tokens). Add documentation.